### PR TITLE
Module Completion Page: Up Next

### PR DESCRIPTION
**Pull request by mslwang**
## Implementation Description
We want to display a card on the module completion page indicating the up-next module.

## Screenshots
image
image
image

## What should reviewers focus on?
- Testing user flow for viewing courses
- Check for different stages of module completion: (Not started, In Progress, Complete)

## Testing Procedure
- What test(s) should we run to make sure your code works?
   - What ways could the input be wrong?
   - Is it secure? Could nefarious users access/break it?

## Things to Note
- Additional dependencies/libraries you've added?
- Things you'd want to know when coming back to the code after a few months

## Checklist
- [x] Ensure code follows style guide by running the linters
- [x] I have updated the documentation or deemed it unnecessary
- [x] I have linked the relevant issue in this PR
- [x] I have requested a review from the PL, as well as other dev(s) (and designers if necessary)

### CODE REVIEW

**Comment by mslwang** [comment]
Remaining todos:
- Fix error with re-direction after clicking Up Next card when subsequent lesson is present
image
- Fix issue where Module Completion page remains after clicking Up Next card if next module does not have any lessons. 

The current assumption elsewhere seems to be that `moduleProgress` length should be the name same as `modules` length, but the following screenshot says otherwise 
image
So if we need to account for people who go to their module page from elsewhere (ie. protection for if you jump from module 3 to module 5 without doing module 4 from the url, probably out of scope of this ticket though), the moduleProgress array won't work correctly since the indices need to line up.

**Comment by mslwang** [comment]
Note: @a4bello, I saw this issue, but I doubt it's one that we need to focus on. When course has been edited after someone has already completed the course, the course completion does not get updated, which messes with `View Course` directing you to the correct module.

https://user-images.githubusercontent.com/53355975/188486851-c1e3bfd1-581f-4a0a-930d-1a1b2ecef540.mov

File: frontend/src/components/module-viewer/UpNext.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  useEffect(() => {
+    if (authenticatedUser) {
+      getModuleProgress({
+        variables: {
+          userId: authenticatedUser.id,
+          courseId: courseID,
+        },
+      });
+    }
+  }, [authenticatedUser, setAuthenticatedUser, getModuleProgress, courseID]);
```
Comment (jfdoming) [inline frontend/src/components/module-viewer/UpNext.tsx:47]: Doesn't look like setAuthenticatedUser is actually used?

**Review by jfdoming — APPROVED** [review]

File: frontend/src/components/module-viewer/UpNext.tsx (hunk 1 above)
Comment (mslwang) [reply frontend/src/components/module-viewer/UpNext.tsx]: Fixed

**Review by jfdoming — APPROVED** [review]

**Review by jfdoming — APPROVED** [review]

**Pull request merged**